packages/cli/lib/artifact-generator.js

```diff
@@ -99,18 +99,42 @@ module.exports = class ArtifactGenerator extends BaseGenerator {
        * Both those properties must be present for this to happen. Optionally,
        * this can be disabled even if the properties are present by setting:
        * this.artifactInfo.disableIndexUpdate = true;
+       *
+       * Multiple indexes / files can be updated by providing an array of
+       * index update objects as follows:
+       * this.artifactInfo.indexesToBeUpdated = [{
+       *   dir: 'directory in which to update index.ts',
+       *   file: 'file to add to index.ts',
+       * }, {dir: '...', file: '...'}]
        */
-      if (
-        this.artifactInfo.outDir &&
-        this.artifactInfo.outFile &&
-        !this.artifactInfo.disableIndexUpdate
-      ) {
-        await updateIndex(this.artifactInfo.outDir, this.artifactInfo.outFile);
-        // Output for users
-        this.log(
-          chalk.green('   update'),
-          `${this.artifactInfo.relPath}/index.ts`,
-        );
+      if (!this.artifactInfo.disableIndexUpdate) {
+        if (
+          !this.artifactInfo.indexesToBeUpdated &&
+          this.artifactInfo.outDir &&
+          this.artifactInfo.outFile
+        ) {
+          this.artifactInfo.indexesToBeUpdated = [
+            {dir: this.artifactInfo.outDir, file: this.artifactInfo.outFile},
+          ];
+        } else {
+          this.artifactInfo.indexesToBeUpdated = [];
+        }
+
+        for (const idx of this.artifactInfo.indexesToBeUpdated) {
+          await updateIndex(idx.dir, idx.file);
+          // Output for users
+          const updateDirRelPath = path.relative(
+            this.artifactInfo.relPath,
+            idx.dir,
+          );
+
+          this.log(
+            chalk.green('   update'),
+            `${this.artifactInfo.relPath}/${updateDirRelPath}${
```

Comment:
I find multi-line templates difficult to read. Have you considered extracting individual bits into local variables?

```ts
const relPath = this.artifactInfo.relPath;
const sep = updateDirRelPath.length > 0 ? '/' : '';
const indexPath = `${relPath}/${updateDirRelPath}${sep}index.ts`;
this.log(chalk.green('   update'), indexPath);
```

I think it would be even better to use Node.js core API [path.join()](https://nodejs.org/api/path.html#path_path_join_paths). Possible caveat: it may use `\` instead of `/` on Windows (don't remember what's the exact behavior), IDK if that would be a good or a bad thing.

```ts
const indexPath = path.join(
  this.artifactInfo.relPath,
  updateDirRelPath,
  'index.ts'
);
this.log(chalk.green('   update'), indexPath);
```

The function handles empty path segments well:

```
$ node
> path.join('/asd', '', 'index.ts')
'/asd/index.ts'
```